packages/pouch-storage/src/PouchStorageTable.ts

```diff
@@ -35,20 +35,18 @@ export type PouchDBSort = Array<
   string | { [propName: string]: 'asc' | 'desc' }
 >;
 
-// We may want to use `PouchDB.Find.ConditionOperators` instead of this, but
-// there are some mismatches in how we use this with the types.
-// https://github.com/deephaven/web-client-ui/issues/1505 to address this
 type PouchFilter = OnlyOneProp<{
   $eq: FilterValue | FilterValue[];
-  $neq: FilterValue | FilterValue[];
+  $ne: FilterValue | FilterValue[];
   $gt: FilterValue | FilterValue[];
   $gte: FilterValue | FilterValue[];
   $lt: FilterValue | FilterValue[];
   $lte: FilterValue | FilterValue[];
-  $regex: RegExp;
-}>;
+  $regex: RegExp | string;
+}> &
+  Omit<PouchDB.Find.ConditionOperators, '$regex'>;
 
-function makePouchFilter(
+export function makePouchFilter(
```

Comment:
@AkshatJawne are there unit tests for this?

Reply:
Created unit tests, but as I mentioned above, there are some issues with PouchDB I believe where Jest cannot understand the import, can speak to you about this offline. 

Essentially, from what I can find, the viable solution is creating a `babel.config.js `and doing the steps outlined here https://stackoverflow.com/questions/54461150/jest-not-parsing-es6-syntaxerror-unexpected-token-import . 

I found this from the comment thread of the original issue outlining difficulties using a different type of PouchDB storage https://stackoverflow.com/questions/64905287/import-error-when-running-jest-tests-that-use-the-pouchdb-asyncstorage-adapter , but again, not sure if that is what we want to do here given scope of code.